### PR TITLE
Use setup-gradle in actions

### PR DESCRIPTION
**Pull request by loosebazooka**
gradle-build-action is deprecated

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -14,21 +14,24 @@ jobs:
...
+      uses: gradle/actions/setup-gradle@v3
+    
+    - name: Build and Check
+      run: ./gradlew build check
+
     - name: Publish to Plugin Portal
-      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
-      with:
-        arguments: publishPlugins -Pgradle.publish.key=$GRADLE_PUBLISH_KEY -Pgradle.publish.secret=$GRADLE_PUBLISH_SECRET
+      run: ./gradlew publishPlugins -Pgradle.publish.key=$GRADLE_PUBLISH_KEY -Pgradle.publish.secret=$GRADLE_PUBLISH_SECRET
```
Comment (mathjeff) [inline .github/workflows/release.yml:34]: It seems weird to me to see an invocation of Gradle making release artifacts without `--no-daemon` - do you know if Gradle daemons would be cleared in-between builds?

Comment (loosebazooka) [reply .github/workflows/release.yml:34]: Oh I'm not familiar enough with what issues might arise. We can use --no-daemon if that makes sense here.

Comment (loosebazooka) [reply .github/workflows/release.yml]: alright, this should just all happen together.

Comment (mathjeff) [reply .github/workflows/release.yml]: talked offline and you mentioned that the machines were isolated so there shouldn't be leftover memory state from previous runs

**Review by mathjeff — APPROVED** [review]

**Pull request merged**